pkg/arvo/app/chat-view.hoon

```diff
@@ -111,11 +128,13 @@
     ?:  ?=(%bound +<.sign-arvo)  [~ this]
     (on-arvo:def wire sign-arvo)
   ::
-  ++  on-save  on-save:def
+  ++  on-save  !>(state)
   ++  on-load  
-    |=  old=*
+    |=  old-vase=vase
     ^-  (quip card _this)
-    :_  this
+    =/  old  ((soft state-0) q.old-vase)
```

Comment:
Cool, this actually works. (:

You could've also unpacked the vase into the `$@(~ state-0)` type instead of softing, but it does the trick.